### PR TITLE
Add unit test for pdf creator

### PR DESCRIPTION
**Pull request by rkondratowicz**
Mostly copied from https://github.com/hmcts/send-letter-consumer-service/blob/master/src/test/java/uk/gov/hmcts/reform/slc/services/steps/getpdf/PdfCreatorTest.java
with slight changes:
- no calls to pdf-service api
- no calls to app insights

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java [hunk 1]
```diff
@@ -33,7 +34,7 @@
 
     private static final Logger log = LoggerFactory.getLogger(LetterService.class);
 
-    private final PdfCreator pdfCreator = new PdfCreator(new DuplexPreparator());
+    private final PdfCreator pdfCreator = new PdfCreator(new DuplexPreparator(), new HTMLToPDFConverter()::convert);
```
Comment (KarolJastrzebski) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: This is already returned in PdfConversionConfig. Why not to use it?

**Comment by rkondratowicz** [comment]
not everything at once

**Comment by doncem** [comment]
yea. but you copied new files over :)

File: src/main/java/uk/gov/hmcts/reform/slc/config/PdfConversionConfig.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package uk.gov.hmcts.reform.slc.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import uk.gov.hmcts.reform.pdf.generator.HTMLToPDFConverter;
+import uk.gov.hmcts.reform.slc.services.steps.getpdf.IHtmlToPdfConverter;
+
+@Configuration
+public class PdfConversionConfig {
```
Comment (KarolJastrzebski) [inline src/main/java/uk/gov/hmcts/reform/slc/config/PdfConversionConfig.java]: Why not to take the burden of calling `new` throughout the app and return configured instance of PdfCreator instead?

File: src/main/java/uk/gov/hmcts/reform/slc/services/steps/getpdf/PdfCreator.java [hunk 3]
```diff
@@ -15,10 +14,11 @@
 public class PdfCreator {
 
     private final DuplexPreparator duplexPreparator;
-    private static HTMLToPDFConverter converter = new HTMLToPDFConverter();
+    private final IHtmlToPdfConverter converter;
```
Comment (KarolJastrzebski) [inline src/main/java/uk/gov/hmcts/reform/slc/services/steps/getpdf/PdfCreator.java]: :+1:

File: src/main/java/uk/gov/hmcts/reform/slc/services/steps/getpdf/IHtmlToPdfConverter.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.hmcts.reform.slc.services.steps.getpdf;
+
+import java.util.Map;
+import java.util.function.BiFunction;
+
+public interface IHtmlToPdfConverter extends BiFunction<byte[], Map<String, Object>, byte[]> {
```
Comment (KarolJastrzebski) [inline src/main/java/uk/gov/hmcts/reform/slc/services/steps/getpdf/IHtmlToPdfConverter.java]: Friendly alias for lambda instance has its benefits. Alternatively, to keep inheritance hierarchy flat and open to other implementations, there are options like:

1.
```java

interface IHtmlToPdfConverter {
    byte[] convert(byte[] template, Map<String, Object> values);
}

class HtmlToPdfFuncAdapter implements IHtmlToPdfConverter {

    private final BiFunction<byte[], Map<String, Object>, byte[]> function;

    HtmlToPdfFuncAdapter(BiFunction<byte[], Map<String, Object>, byte[]> function) {
        this.function = function;
    }

    @Override
    public byte[] convert(byte[] template, Map<String, Object> values) {
        return function.apply(template, values);
    }
}
```
2. Use `BiFunction<byte[], Map<String, Object>, byte[]>` as it is quite technical and descriptive.

File: src/main/java/uk/gov/hmcts/reform/slc/config/PdfConversionConfig.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/slc/config/PdfConversionConfig.java]: `PdfCreator` is a `@Service` and is injected by Spring.

**Pull request merged**